// overlayscrollbars/dist/browser/overlayscrollbars.browser.es5.js
/*!
 * OverlayScrollbars
 * Version: 2.0.0-beta.2
 *
 * Copyright (c) Rene Haas | KingSora.
 * https://github.com/KingSora
 *
 * Released under the MIT license.
 */

var OverlayScrollbarsGlobal = function(r) {
  "use strict";
  function each(r, a) {
    if (g(r)) {
      for (var e = 0; e < r.length; e++) {
        if (false === a(r[e], e, r)) {
          break;
        }
      }
    } else if (r) {
      each(Object.keys(r), (function(e) {
        return a(r[e], e, r);
      }));
    }
    return r;
  }
  function style(r, a) {
    var e = l(a);
    var n = d(a) || e;
    if (n) {
      var t = e ? "" : {};
      if (r) {
        var i = window.getComputedStyle(r, null);
        t = e ? Lr(r, i, a) : a.reduce((function(a, e) {
          a[e] = Lr(r, i, e);
          return a;
        }), t);
      }
      return t;
    }
    r && each(E(a), (function(e) {
      return Hr(r, e, a[e]);
    }));
  }
  function getDefaultExportFromCjs(r) {
    return r && r.v && Object.prototype.hasOwnProperty.call(r, "default") ? r["default"] : r;
  }
  var a = function createCache(r, a) {
    var e = r.o, n = r.u, t = r.g;
    var i = e;
    var v;
    var o = function cacheUpdateContextual(r, a) {
      var e = i;
      var o = r;
      var u = a || (n ? !n(e, o) : e !== o);
      if (u || t) {
        i = o;
        v = e;
      }
      return [ i, u, v ];
    };
    var u = function cacheUpdateIsolated(r) {
      return o(a(i, v), r);
    };
    var f = function getCurrentCache(r) {
      return [ i, !!r, v ];
    };
    return [ a ? u : o, f ];
  };
  var e = Node.ELEMENT_NODE;
  var n = Object.prototype, t = n.toString, i = n.hasOwnProperty;
  var v = function isUndefined(r) {
    return void 0 === r;
  };
  var o = function isNull(r) {
    return null === r;
  };
  var u = function type(r) {
    return v(r) || o(r) ? "" + r : t.call(r).replace(/^\[object (.+)\]$/, "$1").toLowerCase();
  };
  var f = function isNumber(r) {
    return "number" === typeof r;
  };
  var l = function isString(r) {
    return "string" === typeof r;
  };
  var c = function isBoolean(r) {
    return "boolean" === typeof r;
  };
  var s = function isFunction(r) {
    return "function" === typeof r;
  };
  var d = function isArray(r) {
    return Array.isArray(r);
  };
  var h = function isObject(r) {
    return "object" === typeof r && !d(r) && !o(r);
  };
  var g = function isArrayLike(r) {
    var a = !!r && r.length;
    var e = f(a) && a > -1 && a % 1 == 0;
    return d(r) || !s(r) && e ? a > 0 && h(r) ? a - 1 in r : true : false;
  };
  var p = function isPlainObject(r) {
    if (!r || !h(r) || "object" !== u(r)) {
      return false;
    }
    var a;
    var e = "constructor";
    var n = r[e];
    var t = n && n.prototype;
    var o = i.call(r, e);
    var f = t && i.call(t, "isPrototypeOf");
    if (n && !o && !f) {
      return false;
    }
    for (a in r) {}
    return v(a) || i.call(r, a);
  };
  var w = function isHTMLElement(r) {
    var a = HTMLElement;
    return r ? a ? r instanceof a : r.nodeType === e : false;
  };
  var b = function isElement(r) {
    var a = Element;
    return r ? a ? r instanceof a : r.nodeType === e : false;
  };
  var m = function indexOf(r, a, e) {
    return r.indexOf(a, e);
  };
  var y = function push(r, a, e) {
    !e && !l(a) && g(a) ? Array.prototype.push.apply(r, a) : r.push(a);
    return r;
  };
  var S = function from(r) {
    var a = Array.from;
    var e = [];
    if (a && r) {
      return a(r);
    }
    if (r instanceof Set) {
      r.forEach((function(r) {
        y(e, r);
      }));
    } else {
      each(r, (function(r) {
        y(e, r);
      }));
    }
    return e;
  };
  var C = function isEmptyArray(r) {
    return !!r && 0 === r.length;
  };
  var O = function runEachAndClear(r, a, e) {
    var n = function runFn(r) {
      return r && r.apply(void 0, a || []);
    };
    each(r, n);
    !e && (r.length = 0);
  };
  var x = function hasOwnProperty(r, a) {
    return Object.prototype.hasOwnProperty.call(r, a);
  };
  var E = function keys(r) {
    return r ? Object.keys(r) : [];
  };
  var P = function assignDeep(r, a, e, n, t, i, v) {
    var u = [ a, e, n, t, i, v ];
    if (("object" !== typeof r || o(r)) && !s(r)) {
      r = {};
    }
    each(u, (function(a) {
      each(E(a), (function(e) {
        var n = a[e];
        if (r === n) {
          return true;
        }
        var t = d(n);
        if (n && (p(n) || t)) {
          var i = r[e];
          var v = i;
          if (t && !d(i)) {
            v = [];
          } else if (!t && !p(i)) {
            v = {};
          }
          r[e] = assignDeep(v, n);
        } else {
          r[e] = n;
        }
      }));
    }));
    return r;
  };
  var T = function isEmptyObject(r) {
    for (var a in r) {
      return false;
    }
    return true;
  };
  var z = function getSetProp(r, a, e, n) {
    if (v(n)) {
      return e ? e[r] : a;
    }
    e && (l(n) || f(n)) && (e[r] = n);
  };
  var A = function attr(r, a, e) {
    if (v(e)) {
      return r ? r.getAttribute(a) : null;
    }
    r && r.setAttribute(a, e);
  };
  var L = function attrClass(r, a, e, n) {
    if (e) {
      var t = A(r, a) || "";
      var i = new Set(t.split(" "));
      i[n ? "add" : "delete"](e);
      A(r, a, S(i).join(" ").trim());
    }
  };
  var H = function hasAttrClass(r, a, e) {
    var n = A(r, a) || "";
    var t = new Set(n.split(" "));
    return t.has(e);
  };
  var M = function removeAttr(r, a) {
    r && r.removeAttribute(a);
  };
  var R = function scrollLeft(r, a) {
    return z("scrollLeft", 0, r, a);
  };
  var D = function scrollTop(r, a) {
    return z("scrollTop", 0, r, a);
  };
  var I = Element.prototype;
  var k = function find(r, a) {
    var e = [];
    var n = a ? b(a) ? a : null : document;
    return n ? y(e, n.querySelectorAll(r)) : e;
  };
  var V = function findFirst(r, a) {
    var e = a ? b(a) ? a : null : document;
    return e ? e.querySelector(r) : null;
  };
  var j = function is(r, a) {
    if (b(r)) {
      var e = I.matches || I.msMatchesSelector;
      return e.call(r, a);
    }
    return false;
  };
  var B = function contents(r) {
    return r ? S(r.childNodes) : [];
  };
  var F = function parent(r) {
    return r ? r.parentElement : null;
  };
  var q = function closest(r, a) {
    if (b(r)) {
      var e = I.closest;
      if (e) {
        return e.call(r, a);
      }
      do {
        if (j(r, a)) {
          return r;
        }
        r = F(r);
      } while (r);
    }
    return null;
  };
  var U = function liesBetween(r, a, e) {
    var n = r && q(r, a);
    var t = r && V(e, n);
    var i = q(t, a) === n;
    return n && t ? n === r || t === r || i && q(q(r, e), a) !== n : false;
  };
  var N = function before(r, a, e) {
    if (e && r) {
      var n = a;
      var t;
      if (g(e)) {
        t = document.createDocumentFragment();
        each(e, (function(r) {
          if (r === n) {
            n = r.previousSibling;
          }
          t.appendChild(r);
        }));
      } else {
        t = e;
      }
      if (a) {
        if (!n) {
          n = r.firstChild;
        } else if (n !== a) {
          n = n.nextSibling;
        }
      }
      r.insertBefore(t, n || null);
    }
  };
  var Y = function appendChildren(r, a) {
    N(r, null, a);
  };
  var W = function insertBefore(r, a) {
    N(F(r), r, a);
  };
  var G = function insertAfter(r, a) {
    N(F(r), r && r.nextSibling, a);
  };
  var X = function removeElements(r) {
    if (g(r)) {
      each(S(r), (function(r) {
        return removeElements(r);
      }));
    } else if (r) {
      var a = F(r);
      if (a) {
        a.removeChild(r);
      }
    }
  };
  var Z = function createDiv(r) {
    var a = document.createElement("div");
    if (r) {
      A(a, "class", r);
    }
    return a;
  };
  var $ = function createDOM(r) {
    var a = Z();
    a.innerHTML = r.trim();
    return each(B(a), (function(r) {
      return X(r);
    }));
  };
  var J = function firstLetterToUpper(r) {
    return r.charAt(0).toUpperCase() + r.slice(1);
  };
  var K = function getDummyStyle() {
    return Z().style;
  };
  var Q = [ "-webkit-", "-moz-", "-o-", "-ms-" ];
  var rr = [ "WebKit", "Moz", "O", "MS", "webkit", "moz", "o", "ms" ];
  var ar = {};
  var er = {};
  var nr = function cssProperty(r) {
    var a = er[r];
    if (x(er, r)) {
      return a;
    }
    var e = J(r);
    var n = K();
    each(Q, (function(t) {
      var i = t.replace(/-/g, "");
      var v = [ r, t + r, i + e, J(i) + e ];
      return !(a = v.find((function(r) {
        return void 0 !== n[r];
      })));
    }));
    return er[r] = a || "";
  };
  var tr = function jsAPI(r) {
    var a = ar[r] || window[r];
    if (x(ar, r)) {
      return a;
    }
    each(rr, (function(e) {
      a = a || window[e + J(r)];
      return !a;
    }));
    ar[r] = a;
    return a;
  };
  var ir = tr("MutationObserver");
  var vr = tr("IntersectionObserver");
  var or = tr("ResizeObserver");
  var ur = tr("cancelAnimationFrame");
  var fr = tr("requestAnimationFrame");
  var lr = window.setTimeout;
  var cr = window.clearTimeout;
  var sr = function equal(r, a, e, n) {
    if (r && a) {
      var t = true;
      each(e, (function(e) {
        var i = n ? n(r[e]) : r[e];
        var v = n ? n(a[e]) : a[e];
        if (i !== v) {
          t = false;
        }
      }));
      return t;
    }
    return false;
  };
  var dr = function equalWH(r, a) {
    return sr(r, a, [ "w", "h" ]);
  };
  var hr = function equalXY(r, a) {
    return sr(r, a, [ "x", "y" ]);
  };
  var gr = function equalTRBL(r, a) {
    return sr(r, a, [ "t", "r", "b", "l" ]);
  };
  var pr = function equalBCRWH(r, a, e) {
    return sr(r, a, [ "width", "height" ], e && function(r) {
      return Math.round(r);
    });
  };
  var _r = function noop() {};
  var wr = function selfClearTimeout(r) {
    var a;
    var e = r ? lr : fr;
    var n = r ? cr : ur;
    return [ function(t) {
      n(a);
      a = e(t, s(r) ? r() : r);
    }, function() {
      return n(a);
    } ];
  };
  var br = function debounce(r, a) {
    var e;
    var n;
    var t;
    var i = _r;
    var v = a || {}, o = v.p, u = v._, l = v.m;
    var c = function invokeFunctionToDebounce(a) {
      i();
      cr(e);
      e = n = void 0;
      i = _r;
      r.apply(this, a);
    };
    var d = function mergeParms(r) {
      return l && n ? l(n, r) : r;
    };
    var h = function flush() {
      if (i !== _r) {
        c(d(t) || t);
      }
    };
    var g = function debouncedFn() {
      var r = S(arguments);
      var a = s(o) ? o() : o;
      var v = f(a) && a >= 0;
      if (v) {
        var l = s(u) ? u() : u;
        var g = f(l) && l >= 0;
        var p = a > 0 ? lr : fr;
        var w = a > 0 ? cr : ur;
        var b = d(r);
        var m = b || r;
        var y = c.bind(0, m);
        i();
        var C = p(y, a);
        i = function clear() {
          return w(C);
        };
        if (g && !e) {
          e = lr(h, l);
        }
        n = t = m;
      } else {
        c(r);
      }
    };
    g.S = h;
    return g;
  };
  var mr = Math.max;
  var yr = function animationCurrentTime() {
    return performance.now();
  };
  var Sr = function animateNumber(r, a, e, n, t) {
    var i = 0;
    var v = yr();
    var o = Math.max(0, e);
    var u = function frame(e) {
      var u = yr();
      var f = u - v;
      var l = f >= o;
      var c = e ? 1 : 1 - (mr(0, v + o - u) / o || 0);
      var d = (a - r) * (s(t) ? t(c, c * o, 0, 1, o) : c) + r;
      var h = l || 1 === c;
      n && n(d, c, h);
      i = h ? 0 : fr((function() {
        return frame();
      }));
    };
    u();
    return function(r) {
      ur(i);
      r && u(r);
    };
  };
  var Cr = /[^\x20\t\r\n\f]+/g;
  var Or = function classListAction(r, a, e) {
    var n = r && r.classList;
    var t;
    var i = 0;
    var v = false;
    if (n && a && l(a)) {
      var o = a.match(Cr) || [];
      v = o.length > 0;
      while (t = o[i++]) {
        v = !!e(n, t) && v;
      }
    }
    return v;
  };
  var xr = function hasClass(r, a) {
    return Or(r, a, (function(r, a) {
      return r.contains(a);
    }));
  };
  var Er = function removeClass(r, a) {
    Or(r, a, (function(r, a) {
      return r.remove(a);
    }));
  };
  var Pr = function addClass(r, a) {
    Or(r, a, (function(r, a) {
      return r.add(a);
    }));
    return Er.bind(0, r, a);
  };
  var Tr = {
    opacity: 1,
    zindex: 1
  };
  var zr = function parseToZeroOrNumber(r, a) {
    var e = a ? parseFloat(r) : parseInt(r, 10);
    return e === e ? e : 0;
  };
  var Ar = function adaptCSSVal(r, a) {
    return !Tr[r.toLowerCase()] && f(a) ? a + "px" : a;
  };
  var Lr = function getCSSVal(r, a, e) {
    return null != a ? a[e] || a.getPropertyValue(e) : r.style[e];
  };
  var Hr = function setCSSVal(r, a, e) {
    try {
      var n = r.style;
      if (!v(n[a])) {
        n[a] = Ar(a, e);
      } else {
        n.setProperty(a, e);
      }
    } catch (t) {}
  };
  var Mr = function directionIsRTL(r) {
    return "rtl" === style(r, "direction");
  };
  var Rr = function topRightBottomLeft(r, a, e) {
    var n = a ? a + "-" : "";
    var t = e ? "-" + e : "";
    var i = n + "top" + t;
    var v = n + "right" + t;
    var o = n + "bottom" + t;
    var u = n + "left" + t;
    var f = style(r, [ i, v, o, u ]);
    return {
      t: zr(f[i]),
      r: zr(f[v]),
      b: zr(f[o]),
      l: zr(f[u])
    };
  };
  var Dr = Math.round;
  var Ir = {
    w: 0,
    h: 0
  };
  var kr = function windowSize() {
    return {
      w: window.innerWidth,
      h: window.innerHeight
    };
  };
  var Vr = function offsetSize(r) {
    return r ? {
      w: r.offsetWidth,
      h: r.offsetHeight
    } : Ir;
  };
  var jr = function clientSize(r) {
    return r ? {
      w: r.clientWidth,
      h: r.clientHeight
    } : Ir;
  };
  var Br = function scrollSize(r) {
    return r ? {
      w: r.scrollWidth,
      h: r.scrollHeight
    } : Ir;
  };
  var Fr = function fractionalSize(r) {
    var a = parseFloat(style(r, "height")) || 0;
    var e = parseFloat(style(r, "width")) || 0;
    return {
      w: e - Dr(e),
      h: a - Dr(a)
    };
  };
  var qr = function getBoundingClientRect(r) {
    return r.getBoundingClientRect();
  };
  var Ur;
  var Nr = function supportPassiveEvents() {
    if (v(Ur)) {
      Ur = false;
      try {
        window.addEventListener("test", null, Object.defineProperty({}, "passive", {
          get: function get() {
            Ur = true;
          }
        }));
      } catch (r) {}
    }
    return Ur;
  };
  var Yr = function splitEventNames(r) {
    return r.split(" ");
  };
  var Wr = function off(r, a, e, n) {
    each(Yr(a), (function(a) {
      r.removeEventListener(a, e, n);
    }));
  };
  var Gr = function on(r, a, e, n) {
    var t;
    var i = Nr();
    var v = null != (t = i && n && n.C) ? t : i;
    var o = n && n.O || false;
    var u = n && n.P || false;
    var f = [];
    var l = i ? {
      passive: v,
      capture: o
    } : o;
    each(Yr(a), (function(a) {
      var n = u ? function(t) {
        r.removeEventListener(a, n, o);
        e && e(t);
      } : e;
      y(f, Wr.bind(null, r, a, n, o));
      r.addEventListener(a, n, l);
    }));
    return O.bind(0, f);
  };
  var Xr = function stopPropagation(r) {
    return r.stopPropagation();
  };
  var Zr = function preventDefault(r) {
    return r.preventDefault();
  };
  var $r = {
    x: 0,
    y: 0
  };
  var Jr = function absoluteCoordinates(r) {
    var a = r ? qr(r) : 0;
    return a ? {
      x: a.left + window.pageYOffset,
      y: a.top + window.pageXOffset
    } : $r;
  };
  var Kr = function manageListener(r, a) {
    each(d(a) ? a : [ a ], r);
  };
  var Qr = function createEventListenerHub(r) {
    var a = new Map;
    var e = function removeEvent(r, e) {
      if (r) {
        var n = a.get(r);
        Kr((function(r) {
          if (n) {
            n[r ? "delete" : "clear"](r);
          }
        }), e);
      } else {
        a.forEach((function(r) {
          r.clear();
        }));
        a.clear();
      }
    };
    var n = function addEvent(r, n) {
      var t = a.get(r) || new Set;
      a.set(r, t);
      Kr((function(r) {
        r && t.add(r);
      }), n);
      return e.bind(0, r, n);
    };
    var t = function triggerEvent(r, e) {
      var n = a.get(r);
      each(S(n), (function(r) {
        if (e && !C(e)) {
          r.apply(0, e);
        } else {
          r();
        }
      }));
    };
    var i = E(r);
    each(i, (function(a) {
      n(a, r[a]);
    }));
    return [ n, e, t ];
  };
  var ra = function opsStringify(r) {
    return JSON.stringify(r, (function(r, a) {
      if (s(a)) {
        throw new Error;
      }
      return a;
    }));
  };
  var aa = {
    paddingAbsolute: false,
    showNativeOverlaidScrollbars: false,
    update: {
      elementEvents: [ [ "img", "load" ] ],
      debounce: [ 0, 33 ],
      attributes: null,
      ignoreMutation: null
    },
    overflow: {
      x: "scroll",
      y: "scroll"
    },
    scrollbars: {
      theme: "os-theme-dark",
      visibility: "auto",
      autoHide: "never",
      autoHideDelay: 1300,
      dragScroll: true,
      clickScroll: false,
      pointers: [ "mouse", "touch", "pen" ]
    }
  };
  var ea = function getOptionsDiff(r, a) {
    var e = {};
    var n = E(a).concat(E(r));
    each(n, (function(n) {
      var t = r[n];
      var i = a[n];
      if (h(t) && h(i)) {
        P(e[n] = {}, getOptionsDiff(t, i));
        if (T(e[n])) {
          delete e[n];
        }
      } else if (x(a, n) && i !== t) {
        var v = true;
        if (d(t) || d(i)) {
          try {
            if (ra(t) === ra(i)) {
              v = false;
            }
          } catch (o) {}
        }
        if (v) {
          e[n] = i;
        }
      }
    }));
    return e;
  };
  var na = "os-environment";
  var ta = na + "-flexbox-glue";
  var ia = ta + "-max";
  var va = "data-overlayscrollbars";
  var oa = va + "-overflow-x";
  var ua = va + "-overflow-y";
  var fa = "overflowVisible";
  var la = "scrollbarHidden";
  var ca = "updating";
  var sa = "os-padding";
  var da = "os-viewport";
  var ha = da + "-arrange";
  var ga = "os-content";
  var pa = da + "-scrollbar-hidden";
  var _a = "os-overflow-visible";
  var wa = "os-size-observer";
  var ba = wa + "-appear";
  var ma = wa + "-listener";
  var ya = ma + "-scroll";
  var Sa = ma + "-item";
  var Ca = Sa + "-final";
  var Oa = "os-trinsic-observer";
  var xa = "os-scrollbar";
  var Ea = xa + "-rtl";
  var Pa = xa + "-horizontal";
  var Ta = xa + "-vertical";
  var za = xa + "-track";
  var Aa = xa + "-handle";
  var La = xa + "-visible";
  var Ha = xa + "-cornerless";
  var Ma = xa + "-transitionless";
  var Ra = xa + "-interaction";
  var Da = xa + "-unusable";
  var Ia = xa + "-auto-hidden";
  var ka = xa + "-wheel";
  var Va = za + "-interactive";
  var ja = Aa + "-interactive";
  var Ba = {};
  var Fa = function getPlugins() {
    return Ba;
  };
  var qa = function addPlugin(r) {
    var a = [];
    each(d(r) ? r : [ r ], (function(r) {
      var e = E(r);
      each(e, (function(e) {
        y(a, Ba[e] = r[e]);
      }));
    }));
    return a;
  };
  var Ua = {
    exports: {}
  };
  (function(r) {
    function _extends() {
      r.exports = _extends = Object.assign ? Object.assign.bind() : function(r) {
        for (var a = 1; a < arguments.length; a++) {
          var e = arguments[a];
          for (var n in e) {
            if (Object.prototype.hasOwnProperty.call(e, n)) {
              r[n] = e[n];
            }
          }
        }
        return r;
      }, r.exports.v = true, r.exports["default"] = r.exports;
      return _extends.apply(this, arguments);
    }
    r.exports = _extends, r.exports.v = true, r.exports["default"] = r.exports;
  })(Ua);
  var Na = /*@__PURE__*/ getDefaultExportFromCjs(Ua.exports);
  var Ya = {
    boolean: "__TPL_boolean_TYPE__",
    number: "__TPL_number_TYPE__",
    string: "__TPL_string_TYPE__",
    array: "__TPL_array_TYPE__",
    object: "__TPL_object_TYPE__",
    function: "__TPL_function_TYPE__",
    null: "__TPL_null_TYPE__"
  };
  var Wa = function validateRecursive(r, a, e, n) {
    var t = {};
    var i = Na({}, a);
    var o = E(r).filter((function(r) {
      return x(a, r);
    }));
    each(o, (function(o) {
      var f = a[o];
      var c = r[o];
      var s = p(c);
      var h = n ? n + "." : "";
      if (s && p(f)) {
        var g = validateRecursive(c, f, e, h + o), w = g[0], b = g[1];
        t[o] = w;
        i[o] = b;
        each([ i, t ], (function(r) {
          if (T(r[o])) {
            delete r[o];
          }
        }));
      } else if (!s) {
        var m = false;
        var S = [];
        var C = [];
        var O = u(f);
        var x = !d(c) ? [ c ] : c;
        each(x, (function(r) {
          var a;
          each(Ya, (function(e, n) {
            if (e === r) {
              a = n;
            }
          }));
          var e = v(a);
          if (e && l(f)) {
            var n = r.split(" ");
            m = !!n.find((function(r) {
              return r === f;
            }));
            y(S, n);
          } else {
            m = Ya[O] === r;
          }
          y(C, e ? Ya.string : a);
          return !m;
        }));
        if (m) {
          t[o] = f;
        } else if (e) {
          console.warn('The option "' + h + o + "\" wasn't set, because it doesn't accept the type [ " + O.toUpperCase() + ' ] with the value of "' + f + '".\r\n' + "Accepted types are: [ " + C.join(", ").toUpperCase() + " ].\r\n" + (S.length > 0 ? "\r\nValid strings are: [ " + S.join(", ") + " ]." : ""));
        }
        delete i[o];
      }
    }));
    return [ t, i ];
  };
  var Ga = function validateOptions(r, a, e) {
    return Wa(r, a, e);
  };
  var Xa = Ya.number;
  var Za = Ya.boolean;
  var $a = [ Ya.array, Ya.null ];
  var Ja = "hidden scroll visible visible-hidden";
  var Ka = "visible hidden auto";
  var Qa = "never scroll leavemove";
  var re = {
    paddingAbsolute: Za,
    showNativeOverlaidScrollbars: Za,
    update: {
      elementEvents: $a,
      attributes: $a,
      debounce: [ Ya.number, Ya.array, Ya.null ],
      ignoreMutation: [ Ya.function, Ya.null ]
    },
    overflow: {
      x: Ja,
      y: Ja
    },
    scrollbars: {
      theme: [ Ya.string, Ya.null ],
      visibility: Ka,
      autoHide: Qa,
      autoHideDelay: Xa,
      dragScroll: Za,
      clickScroll: Za,
      pointers: [ Ya.array, Ya.null ]
    }
  };
  var ae = "__osOptionsValidationPlugin";
  /* @__PURE__ */  (function() {
    var r;
    return r = {}, r[ae] = {
      T: function _(r, a) {
        var e = Ga(re, r, a), n = e[0], t = e[1];
        return Na({}, t, n);
      }
    }, r;
  })();
  var ee = 3333333;
  var ne = "scroll";
  var te = "__osSizeObserverPlugin";
  var ie = /* @__PURE__ */ function() {
    var r;
    return r = {}, r[te] = {
      T: function _(r, a, e) {
        var n = $('<div class="' + Sa + '" dir="ltr"><div class="' + Sa + '"><div class="' + Ca + '"></div></div><div class="' + Sa + '"><div class="' + Ca + '" style="width: 200%; height: 200%"></div></div></div>');
        Y(r, n);
        Pr(r, ya);
        var t = n[0];
        var i = t.lastChild;
        var v = t.firstChild;
        var o = null == v ? void 0 : v.firstChild;
        var u = Vr(t);
        var f = u;
        var l = false;
        var c;
        var s = function reset() {
          R(v, ee);
          D(v, ee);
          R(i, ee);
          D(i, ee);
        };
        var d = function onResized(r) {
          c = 0;
          if (l) {
            u = f;
            a(true === r);
          }
        };
        var h = function onScroll(r) {
          f = Vr(t);
          l = !r || !dr(f, u);
          if (r) {
            Xr(r);
            if (l && !c) {
              ur(c);
              c = fr(d);
            }
          } else {
            d(false === r);
          }
          s();
        };
        var g = y([], [ Gr(v, ne, h), Gr(i, ne, h) ]);
        style(o, {
          width: ee,
          height: ee
        });
        fr(s);
        return [ e ? h.bind(0, false) : s, g ];
      }
    }, r;
  }();
  var ve = 0;
  var oe = Math.round, ue = Math.abs;
  var fe = function getWindowDPR() {
    var r = window.screen.deviceXDPI || 0;
    var a = window.screen.logicalXDPI || 1;
    return window.devicePixelRatio || r / a;
  };
  var le = function diffBiggerThanOne(r, a) {
    var e = ue(r);
    var n = ue(a);
    return !(e === n || e + 1 === n || e - 1 === n);
  };
  var ce = "__osScrollbarsHidingPlugin";
  var se = /* @__PURE__ */ function() {
    var r;
    return r = {}, r[ce] = {
      A: function _createUniqueViewportArrangeElement(r) {
        var a = r.L, e = r.H, n = r.M;
        var t = !n && !a && (e.x || e.y);
        var i = t ? document.createElement("style") : false;
        if (i) {
          A(i, "id", ha + "-" + ve);
          ve++;
        }
        return i;
      },
      R: function _overflowUpdateSegment(r, a, e, n, t, i, v) {
        var o = function arrangeViewport(a, i, v, o) {
          if (r) {
            var u = t(), f = u.D;
            var l = a.I, c = a.k;
            var s = c.x, d = c.y;
            var h = l.x, g = l.y;
            var p = o ? "paddingRight" : "paddingLeft";
            var w = f[p];
            var b = f.paddingTop;
            var m = i.w + v.w;
            var y = i.h + v.h;
            var S = {
              w: g && d ? g + m - w + "px" : "",
              h: h && s ? h + y - b + "px" : ""
            };
            if (n) {
              var C = n.sheet;
              if (C) {
                var O = C.cssRules;
                if (O) {
                  if (!O.length) {
                    C.insertRule("#" + A(n, "id") + " + ." + ha + "::before {}", 0);
                  }
                  var x = O[0].style;
                  x.width = S.w;
                  x.height = S.h;
                }
              }
            } else {
              style(e, {
                "--os-vaw": S.w,
                "--os-vah": S.h
              });
            }
          }
          return r;
        };
        var u = function undoViewportArrange(n, o, u) {
          if (r) {
            var f = u || i(n);
            var l = t(), c = l.D;
            var s = f.k;
            var d = s.x, h = s.y;
            var g = {};
            var p = function assignProps(r) {
              return each(r.split(" "), (function(r) {
                g[r] = c[r];
              }));
            };
            if (d) {
              p("marginBottom paddingTop paddingBottom");
            }
            if (h) {
              p("marginLeft marginRight paddingLeft paddingRight");
            }
            var w = style(e, E(g));
            Er(e, ha);
            if (!a) {
              g.height = "";
            }
            style(e, g);
            return [ function() {
              v(f, o, r, w);
              style(e, w);
              Pr(e, ha);
            }, f ];
          }
          return [ _r ];
        };
        return [ o, u ];
      },
      V: function _envWindowZoom() {
        var r = {
          w: 0,
          h: 0
        };
        var a = 0;
        return function(e, n, t) {
          var i = kr();
          var v = {
            w: i.w - r.w,
            h: i.h - r.h
          };
          if (0 === v.w && 0 === v.h) {
            return;
          }
          var o = {
            w: ue(v.w),
            h: ue(v.h)
          };
          var u = {
            w: ue(oe(i.w / (r.w / 100))),
            h: ue(oe(i.h / (r.h / 100)))
          };
          var f = fe();
          var l = o.w > 2 && o.h > 2;
          var c = !le(u.w, u.h);
          var s = f !== a && f > 0;
          var d = l && c && s;
          if (d) {
            var h = n(), g = h[0], p = h[1];
            P(e.j, g);
            if (p) {
              t();
            }
          }
          r = i;
          a = f;
        };
      }
    }, r;
  }();
  var de = "__osClickScrollPlugin";
  var he = /* @__PURE__ */ function() {
    var r;
    return r = {}, r[de] = {
      T: function _(r, a, e, n, t) {
        var i = 0;
        var v = _r;
        var o = function animateClickScroll(o) {
          v = Sr(o, o + n * Math.sign(e), 133, (function(e, o, u) {
            r(e);
            var f = a();
            var l = f + n;
            var c = t >= f && t <= l;
            if (u && !c) {
              if (i) {
                animateClickScroll(e);
              } else {
                var s = setTimeout((function() {
                  animateClickScroll(e);
                }), 222);
                v = function clear() {
                  clearTimeout(s);
                };
              }
              i++;
            }
          }));
        };
        o(0);
        return function() {
          return v();
        };
      }
    }, r;
  }();
  var ge;
  var pe = function getNativeScrollbarSize(r, a, e, n) {
    Y(r, a);
    var t = jr(a);
    var i = Vr(a);
    var v = Fr(e);
    n && X(a);
    return {
      x: i.h - t.h + v.h,
      y: i.w - t.w + v.w
    };
  };
  var _e = function getNativeScrollbarsHiding(r) {
    var a = false;
    var e = Pr(r, pa);
    try {
      a = "none" === style(r, nr("scrollbar-width")) || "none" === window.getComputedStyle(r, "::-webkit-scrollbar").getPropertyValue("display");
    } catch (n) {}
    e();
    return a;
  };
  var we = function getRtlScrollBehavior(r, a) {
    var e = "hidden";
    style(r, {
      overflowX: e,
      overflowY: e,
      direction: "rtl"
    });
    R(r, 0);
    var n = Jr(r);
    var t = Jr(a);
    R(r, -999);
    var i = Jr(a);
    return {
      i: n.x === t.x,
      n: t.x !== i.x
    };
  };
  var be = function getFlexboxGlue(r, a) {
    var e = Pr(r, ta);
    var n = qr(r);
    var t = qr(a);
    var i = pr(t, n, true);
    var v = Pr(r, ia);
    var o = qr(r);
    var u = qr(a);
    var f = pr(u, o, true);
    e();
    v();
    return i && f;
  };
  var me = function createEnvironment() {
    var r = document, e = r.body;
    var n = $('<div class="' + na + '"><div></div></div>');
    var t = n[0];
    var i = t.firstChild;
    var v = Qr(), o = v[0], u = v[2];
    var f = a({
      o: pe(e, t, i),
      u: hr
    }, pe.bind(0, e, t, i, true)), l = f[0], c = f[1];
    var s = c(), d = s[0];
    var h = _e(t);
    var g = {
      x: 0 === d.x,
      y: 0 === d.y
    };
    var p = {
      elements: {
        host: null,
        padding: !h,
        viewport: function viewport(r) {
          return h && r === r.ownerDocument.body && r;
        },
        content: false
      },
      scrollbars: {
        slot: true
      },
      cancel: {
        nativeScrollbarsOverlaid: false,
        body: null
      }
    };
    var w = P({}, aa);
    var b = {
      j: d,
      H: g,
      L: h,
      M: "-1" === style(t, "zIndex"),
      B: we(t, i),
      F: be(t, i),
      q: function _addListener(r) {
        return o("_", r);
      },
      U: P.bind(0, {}, p),
      N: function _setDefaultInitialization(r) {
        P(p, r);
      },
      Y: P.bind(0, {}, w),
      W: function _setDefaultOptions(r) {
        P(w, r);
      },
      G: P({}, p),
      X: P({}, w)
    };
    M(t, "style");
    X(t);
    if (!h && (!g.x || !g.y)) {
      var m;
      window.addEventListener("resize", (function() {
        var r = Fa()[ce];
        m = m || r && r.V();
        m && m(b, l, u.bind(0, "_"));
      }));
    }
    return b;
  };
  var ye = function getEnvironment() {
    if (!ge) {
      ge = me();
    }
    return ge;
  };
  var Se = function resolveInitialization(r, a) {
    return s(r) ? r.apply(0, a) : r;
  };
  var Ce = function staticInitializationElement(r, a, e, n) {
    var t = v(n) ? e : n;
    var i = Se(t, r);
    return i || a.apply(0, r);
  };
  var Oe = function dynamicInitializationElement(r, a, e, n) {
    var t = v(n) ? e : n;
    var i = Se(t, r);
    return !!i && (w(i) ? i : a.apply(0, r));
  };
  var xe = function cancelInitialization(r, a, e) {
    var n = e || {}, t = n.nativeScrollbarsOverlaid, i = n.body;
    var u = ye(), f = u.H, l = u.L;
    var c = a.nativeScrollbarsOverlaid, s = a.body;
    var d = null != t ? t : c;
    var h = v(i) ? s : i;
    var g = (f.x || f.y) && d;
    var p = r && (o(h) ? !l : h);
    return !!g || !!p;
  };
  var Ee = new WeakMap;
  var Pe = function addInstance(r, a) {
    Ee.set(r, a);
  };
  var Te = function removeInstance(r) {
    Ee.delete(r);
  };
  var ze = function getInstance(r) {
    return Ee.get(r);
  };
  var Ae = function getPropByPath(r, a) {
    return r ? a.split(".").reduce((function(r, a) {
      return r && x(r, a) ? r[a] : void 0;
    }), r) : void 0;
  };
  var Le = function createOptionCheck(r, a, e) {
    return function(n) {
      return [ Ae(r, n), e || void 0 !== Ae(a, n) ];
    };
  };
  var He = function createState(r) {
    var a = r;
    return [ function() {
      return a;
    }, function(r) {
      a = P({}, a, r);
    } ];
  };
  var Me = "tabindex";
  var Re = Z.bind(0, "");
  var De = function unwrap(r) {
    Y(F(r), B(r));
    X(r);
  };
  var Ie = function addDataAttrHost(r, a) {
    A(r, va, a);
    return M.bind(0, r, va);
  };
  var ke = function createStructureSetupElements(r) {
    var a = ye();
    var e = a.U, n = a.L;
    var t = Fa()[ce];
    var i = t && t.A;
    var v = e(), o = v.elements;
    var u = o.host, f = o.viewport, l = o.padding, c = o.content;
    var s = w(r);
    var d = s ? {} : r;
    var h = d.elements;
    var g = h || {}, p = g.host, b = g.padding, S = g.viewport, C = g.content;
    var x = s ? r : d.target;
    var P = j(x, "textarea");
    var T = x.ownerDocument;
    var z = x === T.body;
    var R = T.defaultView;
    var D = Ce.bind(0, [ x ]);
    var I = Oe.bind(0, [ x ]);
    var k = D(Re, f, S);
    var V = k === x;
    var q = V && z;
    var U = T.activeElement;
    var N = !V && R.top === R && U === x;
    var Z = {
      Z: x,
      $: P ? D(Re, u, p) : x,
      J: k,
      K: !V && I(Re, l, b),
      rr: !V && I(Re, c, C),
      ar: !V && !n && i && i(a),
      er: q ? T.documentElement : k,
      nr: q ? T : k,
      tr: R,
      ir: T,
      vr: P,
      ur: z,
      lr: s,
      cr: V,
      sr: function _viewportHasClass(r, a) {
        return V ? H(k, va, a) : xr(k, r);
      },
      dr: function _viewportAddRemoveClass(r, a, e) {
        return V ? L(k, va, a, e) : (e ? Pr : Er)(k, r);
      }
    };
    var $ = E(Z).reduce((function(r, a) {
      var e = Z[a];
      return y(r, e && !F(e) ? e : false);
    }), []);
    var J = function elementIsGenerated(r) {
      return r ? m($, r) > -1 : null;
    };
    var K = Z.Z, Q = Z.$, rr = Z.K, ar = Z.J, er = Z.rr, nr = Z.ar;
    var tr = [];
    var ir = P && J(Q);
    var vr = P ? K : B([ er, ar, rr, Q, K ].find((function(r) {
      return false === J(r);
    })));
    var or = er || ar;
    var ur = function appendElements() {
      var r = Ie(Q, V ? "viewport" : "host");
      var a = Pr(rr, sa);
      var e = Pr(ar, !V && da);
      var t = Pr(er, ga);
      var i = z ? Pr(F(x), pa) : _r;
      if (ir) {
        G(K, Q);
        y(tr, (function() {
          G(Q, K);
          X(Q);
        }));
      }
      Y(or, vr);
      Y(Q, rr);
      Y(rr || Q, !V && ar);
      Y(ar, er);
      y(tr, (function() {
        i();
        r();
        M(ar, oa);
        M(ar, ua);
        if (J(er)) {
          De(er);
        }
        if (J(ar)) {
          De(ar);
        }
        if (J(rr)) {
          De(rr);
        }
        a();
        e();
        t();
      }));
      if (n && !V) {
        y(tr, Er.bind(0, ar, pa));
      }
      if (nr) {
        W(ar, nr);
        y(tr, X.bind(0, nr));
      }
      if (N) {
        var v = A(ar, Me);
        A(ar, Me, "-1");
        ar.focus();
        var o = Gr(T, "pointerdown keydown", (function() {
          v ? A(ar, Me, v) : M(ar, Me);
          o();
        }));
      } else if (U && U.focus) {
        U.focus();
      }
      vr = 0;
    };
    return [ Z, ur, O.bind(0, tr) ];
  };
  var Ve = function createTrinsicUpdateSegment(r, a) {
    var e = r.rr;
    var n = a[0];
    return function(r) {
      var a = ye(), t = a.F;
      var i = n(), v = i.hr;
      var o = r.gr;
      var u = (e || !t) && o;
      if (u) {
        style(e, {
          height: v ? "" : "100%"
        });
      }
      return {
        pr: u,
        _r: u
      };
    };
  };
  var je = function createPaddingUpdateSegment(r, e) {
    var n = e[0], t = e[1];
    var i = r.$, v = r.K, o = r.J, u = r.cr;
    var f = a({
      u: gr,
      o: Rr()
    }, Rr.bind(0, i, "padding", "")), l = f[0], c = f[1];
    return function(r, a, e) {
      var i = c(e), f = i[0], s = i[1];
      var d = ye(), h = d.L, g = d.F;
      var p = n(), w = p.wr;
      var b = r.pr, m = r._r, y = r.br;
      var S = a("paddingAbsolute"), C = S[0], O = S[1];
      var x = !g && m;
      if (b || s || x) {
        var E = l(e);
        f = E[0];
        s = E[1];
      }
      var T = !u && (O || y || s);
      if (T) {
        var z = !C || !v && !h;
        var A = f.r + f.l;
        var L = f.t + f.b;
        var H = {
          marginRight: z && !w ? -A : 0,
          marginBottom: z ? -L : 0,
          marginLeft: z && w ? -A : 0,
          top: z ? -f.t : 0,
          right: z ? w ? -f.r : "auto" : 0,
          left: z ? w ? "auto" : -f.l : 0,
          width: z ? "calc(100% + " + A + "px)" : ""
        };
        var M = {
          paddingTop: z ? f.t : 0,
          paddingRight: z ? f.r : 0,
          paddingBottom: z ? f.b : 0,
          paddingLeft: z ? f.l : 0
        };
        style(v || o, H);
        style(o, M);
        t({
          K: f,
          mr: !z,
          D: v ? M : P({}, H, M)
        });
      }
      return {
        yr: T
      };
    };
  };
  var Be = Math.max;
  var Fe = Be.bind(0, 0);
  var qe = "visible";
  var Ue = "hidden";
  var Ne = 42;
  var Ye = {
    u: dr,
    o: {
      w: 0,
      h: 0
    }
  };
  var We = {
    u: hr,
    o: {
      x: Ue,
      y: Ue
    }
  };
  var Ge = function getOverflowAmount(r, a) {
    var e = window.devicePixelRatio % 1 !== 0 ? 1 : 0;
    var n = {
      w: Fe(r.w - a.w),
      h: Fe(r.h - a.h)
    };
    return {
      w: n.w > e ? n.w : 0,
      h: n.h > e ? n.h : 0
    };
  };
  var Xe = function conditionalClass(r, a, e) {
    return e ? Pr(r, a) : Er(r, a);
  };
  var Ze = function overflowIsVisible(r) {
    return 0 === r.indexOf(qe);
  };
  var $e = function createOverflowUpdateSegment(r, e) {
    var n = e[0], t = e[1];
    var i = r.$, v = r.K, o = r.J, u = r.ar, f = r.cr, l = r.dr, c = r.ur, s = r.tr;
    var d = ye(), h = d.j, g = d.F, p = d.L, w = d.H;
    var b = Fa()[ce];
    var m = !f && !p && (w.x || w.y);
    var y = c && f;
    var S = a(Ye, Fr.bind(0, o)), C = S[0], O = S[1];
    var x = a(Ye, Br.bind(0, o)), E = x[0], P = x[1];
    var T = a(Ye), z = T[0], H = T[1];
    var M = a(Ye), R = M[0], D = M[1];
    var I = a(We), k = I[0];
    var V = function fixFlexboxGlue(r, a) {
      style(o, {
        height: ""
      });
      if (a) {
        var e = n(), t = e.mr, v = e.K;
        var u = r.Sr, f = r.I;
        var l = Fr(i);
        var c = jr(i);
        var s = "content-box" === style(o, "boxSizing");
        var d = t || s ? v.b + v.t : 0;
        var h = !(w.x && s);
        style(o, {
          height: c.h + l.h + (u.x && h ? f.x : 0) - d
        });
      }
    };
    var j = function getViewportOverflowState(r, a) {
      var e = !p && !r ? Ne : 0;
      var n = function getStatePerAxis(r, n, t) {
        var i = style(o, r);
        var v = a ? a[r] : i;
        var u = "scroll" === v;
        var f = n ? e : t;
        var l = u && !p ? f : 0;
        var c = n && !!e;
        return [ i, u, l, c ];
      };
      var t = n("overflowX", w.x, h.x), i = t[0], v = t[1], u = t[2], f = t[3];
      var l = n("overflowY", w.y, h.y), c = l[0], s = l[1], d = l[2], g = l[3];
      return {
        Cr: {
          x: i,
          y: c
        },
        Sr: {
          x: v,
          y: s
        },
        I: {
          x: u,
          y: d
        },
        k: {
          x: f,
          y: g
        }
      };
    };
    var B = function setViewportOverflowState(r, a, e, n) {
      var t = function setAxisOverflowStyle(r, a) {
        var e = Ze(r);
        var n = a && e && r.replace(qe + "-", "") || "";
        return [ a && !e ? r : "", Ze(n) ? "hidden" : n ];
      };
      var i = t(e.x, a.x), v = i[0], o = i[1];
      var u = t(e.y, a.y), f = u[0], l = u[1];
      n.overflowX = o && f ? o : v;
      n.overflowY = l && v ? l : f;
      return j(r, n);
    };
    var F = function hideNativeScrollbars(r, a, e, t) {
      var i = r.I, v = r.k;
      var o = v.x, u = v.y;
      var f = i.x, l = i.y;
      var c = n(), s = c.D;
      var d = a ? "marginLeft" : "marginRight";
      var h = a ? "paddingLeft" : "paddingRight";
      var g = s[d];
      var p = s.marginBottom;
      var w = s[h];
      var b = s.paddingBottom;
      t.width = "calc(100% + " + (l + -1 * g) + "px)";
      t[d] = -l + g;
      t.marginBottom = -f + p;
      if (e) {
        t[h] = w + (u ? l : 0);
        t.paddingBottom = b + (o ? f : 0);
      }
    };
    var q = b ? b.R(m, g, o, u, n, j, F) : [ function() {
      return m;
    }, function() {
      return [ _r ];
    } ], U = q[0], N = q[1];
    return function(r, a, e) {
      var u = r.pr, c = r.Or, d = r._r, h = r.yr, b = r.gr, m = r.br;
      var S = n(), x = S.hr, T = S.wr;
      var M = a("showNativeOverlaidScrollbars"), I = M[0], q = M[1];
      var Y = a("overflow"), W = Y[0], G = Y[1];
      var X = I && w.x && w.y;
      var Z = !f && !g && (u || d || c || q || b);
      var $ = Ze(W.x);
      var J = Ze(W.y);
      var K = $ || J;
      var Q = O(e);
      var rr = P(e);
      var ar = H(e);
      var er = D(e);
      var nr;
      if (q && p) {
        l(pa, la, !X);
      }
      if (Z) {
        nr = j(X);
        V(nr, x);
      }
      if (u || h || d || m || q) {
        if (K) {
          l(_a, fa, false);
        }
        var tr = N(X, T, nr), ir = tr[0], vr = tr[1];
        var or = Q = C(e), ur = or[0], fr = or[1];
        var lr = rr = E(e), cr = lr[0], sr = lr[1];
        var dr = jr(o);
        var hr = cr;
        var gr = dr;
        ir();
        if ((sr || fr || q) && vr && !X && U(vr, cr, ur, T)) {
          gr = jr(o);
          hr = Br(o);
        }
        var pr = {
          w: Fe(Be(cr.w, hr.w) + ur.w),
          h: Fe(Be(cr.h, hr.h) + ur.h)
        };
        var _r = {
          w: Fe(y ? s.innerWidth : gr.w + Fe(dr.w - cr.w) + ur.w),
          h: Fe(y ? s.innerHeight : gr.h + Fe(dr.h - cr.h) + ur.h)
        };
        er = R(_r);
        ar = z(Ge(pr, _r), e);
      }
      var wr = er, br = wr[0], mr = wr[1];
      var yr = ar, Sr = yr[0], Cr = yr[1];
      var Or = rr, xr = Or[0], Er = Or[1];
      var Pr = Q, Tr = Pr[0], zr = Pr[1];
      var Ar = {
        x: Sr.w > 0,
        y: Sr.h > 0
      };
      var Lr = $ && J && (Ar.x || Ar.y) || $ && Ar.x && !Ar.y || J && Ar.y && !Ar.x;
      if (h || m || zr || Er || mr || Cr || G || q || Z) {
        var Hr = {
          marginRight: 0,
          marginBottom: 0,
          marginLeft: 0,
          width: "",
          overflowY: "",
          overflowX: ""
        };
        var Mr = B(X, Ar, W, Hr);
        var Rr = U(Mr, xr, Tr, T);
        if (!f) {
          F(Mr, T, Rr, Hr);
        }
        if (Z) {
          V(Mr, x);
        }
        if (f) {
          A(i, oa, Hr.overflowX);
          A(i, ua, Hr.overflowY);
        } else {
          style(o, Hr);
        }
      }
      L(i, va, fa, Lr);
      Xe(v, _a, Lr);
      !f && Xe(o, _a, K);
      var Dr = k(j(X).Cr), Ir = Dr[0], kr = Dr[1];
      t({
        Cr: Ir,
        Er: {
          x: br.w,
          y: br.h
        },
        Pr: {
          x: Sr.w,
          y: Sr.h
        },
        Tr: Ar
      });
      return {
        zr: kr,
        Ar: mr,
        Lr: Cr
      };
    };
  };
  var Je = function prepareUpdateHints(r, a, e) {
    var n = {};
    var t = a || {};
    var i = E(r).concat(E(t));
    each(i, (function(a) {
      var i = r[a];
      var v = t[a];
      n[a] = !!(e || i || v);
    }));
    return n;
  };
  var Ke = function createStructureSetupUpdate(r, a) {
    var e = r.Z, n = r.J, t = r.dr, i = r.cr;
    var v = ye(), o = v.L, u = v.H, f = v.F;
    var l = !o && (u.x || u.y);
    var c = [ Ve(r, a), je(r, a), $e(r, a) ];
    return function(r, a, v) {
      var o = Je(P({
        pr: false,
        yr: false,
        br: false,
        gr: false,
        Ar: false,
        Lr: false,
        zr: false,
        Or: false,
        _r: false
      }, a), {}, v);
      var u = l || !f;
      var s = u && R(n);
      var d = u && D(n);
      t("", ca, true);
      var h = o;
      each(c, (function(a) {
        h = Je(h, a(h, r, !!v) || {}, v);
      }));
      R(n, s);
      D(n, d);
      t("", ca);
      if (!i) {
        R(e, 0);
        D(e, 0);
      }
      return h;
    };
  };
  var Qe = function createEventContentChange(r, a, e) {
    var n;
    var t = false;
    var i = function destroy() {
      t = true;
    };
    var v = function updateElements(i) {
      if (e) {
        var v = e.reduce((function(a, e) {
          if (e) {
            var n = e[0], t = e[1];
            var v = t && n && (i ? i(n) : k(n, r));
            if (v && v.length && t && l(t)) {
              y(a, [ v, t.trim() ], true);
            }
          }
          return a;
        }), []);
        each(v, (function(e) {
          return each(e[0], (function(i) {
            var v = e[1];
            var o = n.get(i) || [];
            var u = r.contains(i);
            if (u) {
              var f = Gr(i, v, (function(r) {
                if (t) {
                  f();
                  n.delete(i);
                } else {
                  a(r);
                }
              }));
              n.set(i, y(o, f));
            } else {
              O(o);
              n.delete(i);
            }
          }));
        }));
      }
    };
    if (e) {
      n = new WeakMap;
      v();
    }
    return [ i, v ];
  };
  var rn = function createDOMObserver(r, a, e, n) {
    var t = false;
    var i = n || {}, v = i.Hr, o = i.Mr, u = i.Rr, f = i.Dr, c = i.Ir, s = i.kr;
    var d = br((function() {
      if (t) {
        e(true);
      }
    }), {
      p: 33,
      _: 99
    });
    var h = Qe(r, d, u), g = h[0], p = h[1];
    var w = v || [];
    var b = o || [];
    var O = w.concat(b);
    var x = function observerCallback(t, i) {
      var v = c || _r;
      var o = s || _r;
      var u = new Set;
      var d = new Set;
      var h = false;
      var g = false;
      each(t, (function(e) {
        var t = e.attributeName, i = e.target, c = e.type, s = e.oldValue, p = e.addedNodes, w = e.removedNodes;
        var y = "attributes" === c;
        var S = "childList" === c;
        var C = r === i;
        var O = y && l(t) ? A(i, t) : 0;
        var x = 0 !== O && s !== O;
        var E = m(b, t) > -1 && x;
        if (a && (S || !C)) {
          var P = !y;
          var T = y && x;
          var z = T && f && j(i, f);
          var L = z ? !v(i, t, s, O) : P || T;
          var H = L && !o(e, !!z, r, n);
          each(p, (function(r) {
            return u.add(r);
          }));
          each(w, (function(r) {
            return u.add(r);
          }));
          g = g || H;
        }
        if (!a && C && x && !v(i, t, s, O)) {
          d.add(t);
          h = h || E;
        }
      }));
      if (u.size > 0) {
        p((function(r) {
          return S(u).reduce((function(a, e) {
            y(a, k(r, e));
            return j(e, r) ? y(a, e) : a;
          }), []);
        }));
      }
      if (a) {
        !i && g && e(false);
        return [ false ];
      }
      if (d.size > 0 || h) {
        var w = [ S(d), h ];
        !i && e.apply(0, w);
        return w;
      }
    };
    var E = new ir((function(r) {
      return x(r);
    }));
    E.observe(r, {
      attributes: true,
      attributeOldValue: true,
      attributeFilter: O,
      subtree: a,
      childList: a,
      characterData: a
    });
    t = true;
    return [ function() {
      if (t) {
        g();
        E.disconnect();
        t = false;
      }
    }, function() {
      if (t) {
        d.S();
        var r = E.takeRecords();
        return !C(r) && x(r, true);
      }
    } ];
  };
  var an = 3333333;
  var en = function domRectHasDimensions(r) {
    return r && (r.height || r.width);
  };
  var nn = function createSizeObserver(r, e, n) {
    var t = n || {}, i = t.Vr, v = void 0 === i ? false : i, o = t.jr, u = void 0 === o ? false : o;
    var f = Fa()[te];
    var l = ye(), s = l.B;
    var g = $('<div class="' + wa + '"><div class="' + ma + '"></div></div>');
    var p = g[0];
    var w = p.firstChild;
    var b = Mr.bind(0, r);
    var m = a({
      o: void 0,
      g: true,
      u: function _equal(r, a) {
        return !(!r || !en(r) && en(a));
      }
    }), S = m[0];
    var C = function onSizeChangedCallbackProxy(r) {
      var a = d(r) && r.length > 0 && h(r[0]);
      var n = !a && c(r[0]);
      var t = false;
      var i = false;
      var o = true;
      if (a) {
        var u = S(r.pop().contentRect), f = u[0], l = u[2];
        var g = en(f);
        var w = en(l);
        t = !l || !g;
        i = !w && g;
        o = !t;
      } else if (n) {
        o = r[1];
      } else {
        i = true === r;
      }
      if (v && o) {
        var b = n ? r[0] : Mr(p);
        R(p, b ? s.n ? -an : s.i ? 0 : an : an);
        D(p, an);
      }
      if (!t) {
        e({
          pr: !n,
          Br: n ? r : void 0,
          jr: !!i
        });
      }
    };
    var x = [];
    var E = u ? C : false;
    return [ function() {
      O(x);
      X(p);
    }, function() {
      if (or) {
        var e = new or(C);
        e.observe(w);
        y(x, (function() {
          e.disconnect();
        }));
      } else if (f) {
        var n = f.T(w, C, u), t = n[0], i = n[1];
        E = t;
        y(x, i);
      }
      if (v) {
        var o = a({
          o: !b()
        }, b), l = o[0];
        y(x, Gr(p, "scroll", (function(r) {
          var a = l();
          var e = a[0], n = a[1];
          if (n) {
            Er(w, "ltr rtl");
            if (e) {
              Pr(w, "rtl");
            } else {
              Pr(w, "ltr");
            }
            C(a);
          }
          Xr(r);
        })));
      }
      if (E) {
        Pr(p, ba);
        y(x, Gr(p, "animationstart", E, {
          P: !!or
        }));
      }
      if (or || f) {
        Y(r, p);
      }
    } ];
  };
  var tn = function isHeightIntrinsic(r) {
    return 0 === r.h || r.isIntersecting || r.intersectionRatio > 0;
  };
  var vn = function createTrinsicObserver(r, e) {
    var n;
    var t = Z(Oa);
    var i = [];
    var v = a({
      o: false
    }), o = v[0];
    var u = function triggerOnTrinsicChangedCallback(r, a) {
      if (r) {
        var n = o(tn(r));
        var t = n[1];
        if (t) {
          !a && e(n);
          return [ n ];
        }
      }
    };
    var f = function intersectionObserverCallback(r, a) {
      if (r && r.length > 0) {
        return u(r.pop(), a);
      }
    };
    return [ function() {
      O(i);
      X(t);
    }, function() {
      if (vr) {
        n = new vr((function(r) {
          return f(r);
        }), {
          root: r
        });
        n.observe(t);
        y(i, (function() {
          n.disconnect();
        }));
      } else {
        var a = function onSizeChanged() {
          var r = Vr(t);
          u(r);
        };
        var e = nn(t, a), v = e[0], o = e[1];
        y(i, v);
        o();
        a();
      }
      Y(r, t);
    }, function() {
      if (n) {
        return f(n.takeRecords(), true);
      }
    } ];
  };
  var un = "[" + va + "]";
  var fn = "." + da;
  var ln = [ "tabindex" ];
  var cn = [ "wrap", "cols", "rows" ];
  var sn = [ "id", "class", "style", "open" ];
  var dn = function createStructureSetupObservers(r, e, n) {
    var t;
    var i;
    var v;
    var o = r.$, u = r.J, c = r.rr, h = r.vr, g = r.cr, p = r.sr, w = r.dr;
    var b = ye(), S = b.F;
    var C = a({
      u: dr,
      o: {
        w: 0,
        h: 0
      }
    }, (function() {
      var r = p(_a, fa);
      var a = p(ha, "");
      var e = a && R(u);
      var n = a && D(u);
      w(_a, fa);
      w(ha, "");
      w("", ca, true);
      var t = Br(c);
      var i = Br(u);
      var v = Fr(u);
      w(_a, fa, r);
      w(ha, "", a);
      w("", ca);
      R(u, e);
      D(u, n);
      return {
        w: i.w + t.w + v.w,
        h: i.h + t.h + v.h
      };
    })), O = C[0];
    var x = h ? cn : sn.concat(cn);
    var T = br(n, {
      p: function _timeout() {
        return t;
      },
      _: function _maxDelay() {
        return i;
      },
      m: function _mergeParams(r, a) {
        var e = r[0];
        var n = a[0];
        return [ E(e).concat(E(n)).reduce((function(r, a) {
          r[a] = e[a] || n[a];
          return r;
        }), {}) ];
      }
    });
    var z = function updateViewportAttrsFromHost(r) {
      each(r || ln, (function(r) {
        if (m(ln, r) > -1) {
          var a = A(o, r);
          if (l(a)) {
            A(u, r, a);
          } else {
            M(u, r);
          }
        }
      }));
    };
    var L = function onTrinsicChanged(r, a) {
      var t = r[0], i = r[1];
      var v = {
        gr: i
      };
      e({
        hr: t
      });
      !a && n(v);
      return v;
    };
    var H = function onSizeChanged(r) {
      var a = r.pr, t = r.Br, i = r.jr;
      var v = !a || i ? n : T;
      var o = false;
      if (t) {
        var u = t[0], f = t[1];
        o = f;
        e({
          wr: u
        });
      }
      v({
        pr: a,
        br: o
      });
    };
    var I = function onContentMutation(r, a) {
      var e = O(), t = e[1];
      var i = {
        _r: t
      };
      var v = r ? n : T;
      if (t) {
        !a && v(i);
      }
      return i;
    };
    var k = function onHostMutation(r, a, e) {
      var n = {
        Or: a
      };
      if (a) {
        !e && T(n);
      } else if (!g) {
        z(r);
      }
      return n;
    };
    var V = c || !S ? vn(o, L) : [ _r, _r, _r ], j = V[0], B = V[1], F = V[2];
    var N = !g ? nn(o, H, {
      jr: true,
      Vr: true
    }) : [ _r, _r ], Y = N[0], W = N[1];
    var G = rn(o, false, k, {
      Mr: sn,
      Hr: sn.concat(ln)
    }), X = G[0], Z = G[1];
    var $ = g && or && new or(H.bind(0, {
      pr: true
    }));
    $ && $.observe(o);
    z();
    return [ function() {
      j();
      Y();
      v && v[0]();
      $ && $.disconnect();
      X();
    }, function() {
      W();
      B();
    }, function() {
      var r = {};
      var a = Z();
      var e = F();
      var n = v && v[1]();
      if (a) {
        P(r, k.apply(0, y(a, true)));
      }
      if (e) {
        P(r, L.apply(0, y(e, true)));
      }
      if (n) {
        P(r, I.apply(0, y(n, true)));
      }
      return r;
    }, function(r) {
      var a = r("update.ignoreMutation"), e = a[0];
      var n = r("update.attributes"), o = n[0], l = n[1];
      var h = r("update.elementEvents"), p = h[0], w = h[1];
      var b = r("update.debounce"), m = b[0], y = b[1];
      var S = w || l;
      var C = function ignoreMutationFromOptions(r) {
        return s(e) && e(r);
      };
      if (S) {
        if (v) {
          v[1]();
          v[0]();
        }
        v = rn(c || u, true, I, {
          Hr: x.concat(o || []),
          Rr: p,
          Dr: un,
          kr: function _ignoreContentChange(r, a) {
            var e = r.target, n = r.attributeName;
            var t = !a && n && !g ? U(e, un, fn) : false;
            return t || !!q(e, "." + xa) || !!C(r);
          }
        });
      }
      if (y) {
        T.S();
        if (d(m)) {
          var O = m[0];
          var E = m[1];
          t = f(O) && O;
          i = f(E) && E;
        } else if (f(m)) {
          t = m;
          i = false;
        } else {
          t = false;
          i = false;
        }
      }
    } ];
  };
  var hn = {
    x: 0,
    y: 0
  };
  var gn = {
    K: {
      t: 0,
      r: 0,
      b: 0,
      l: 0
    },
    mr: false,
    D: {
      marginRight: 0,
      marginBottom: 0,
      marginLeft: 0,
      paddingTop: 0,
      paddingRight: 0,
      paddingBottom: 0,
      paddingLeft: 0
    },
    Er: hn,
    Pr: hn,
    Cr: {
      x: "hidden",
      y: "hidden"
    },
    Tr: {
      x: false,
      y: false
    },
    hr: false,
    wr: false
  };
  var pn = function createStructureSetup(r, a) {
    var e = Le(a, {});
    var n = He(gn);
    var t = Qr(), i = t[0], v = t[1], o = t[2];
    var u = n[0], f = n[1];
    var l = ke(r), c = l[0], s = l[1], d = l[2];
    var h = Ke(c, n);
    var g = function triggerUpdateEvent(r, a, e) {
      var n = E(r).some((function(a) {
        return r[a];
      }));
      var t = n || !T(a) || e;
      if (t) {
        o("u", [ r, a, e ]);
      }
      return t;
    };
    var p = dn(c, f, (function(r) {
      return g(h(e, r), {}, false);
    })), w = p[0], b = p[1], m = p[2], y = p[3];
    var S = u.bind(0);
    S.Fr = function(r) {
      return i("u", r);
    };
    S.qr = function() {
      var r = c.Z, a = c.J;
      var e = R(r);
      var n = D(r);
      b();
      s();
      R(a, e);
      D(a, n);
    };
    S.Ur = c;
    return [ function(r, e) {
      var n = Le(a, r, e);
      y(n);
      return g(h(n, m(), e), r, !!e);
    }, S, function() {
      v();
      w();
      d();
    } ];
  };
  var _n = Math.round;
  var wn = function getScale(r) {
    var a = qr(r), e = a.width, n = a.height;
    var t = Vr(r), i = t.w, v = t.h;
    return {
      x: _n(e) / i || 1,
      y: _n(n) / v || 1
    };
  };
  var bn = function continuePointerDown(r, a, e) {
    var n = a.scrollbars;
    var t = r.button, i = r.isPrimary, v = r.pointerType;
    var o = n.pointers;
    return 0 === t && i && n[e ? "dragScroll" : "clickScroll"] && (o || []).includes(v);
  };
  var mn = function createRootClickStopPropagationEvents(r, a) {
    return Gr(r, "mousedown", Gr.bind(0, a, "click", Xr, {
      P: true,
      O: true
    }), {
      O: true
    });
  };
  var yn = function createInteractiveScrollEvents(r, a, e, n, t, i) {
    var v = ye(), o = v.B;
    var u = e.Nr, f = e.Yr, l = e.Wr;
    var c = "scroll" + (i ? "Left" : "Top");
    var s = "client" + (i ? "X" : "Y");
    var d = i ? "width" : "height";
    var h = i ? "left" : "top";
    var g = i ? "w" : "h";
    var p = i ? "x" : "y";
    var w = function createRelativeHandleMove(r, a) {
      return function(e) {
        var v = t(), s = v.Pr;
        var d = Vr(f)[g] - Vr(u)[g];
        var h = a * e / d;
        var w = h * s[p];
        var b = Mr(l);
        var m = b && i ? o.n || o.i ? 1 : -1 : 1;
        n[c] = r + w * m;
      };
    };
    return Gr(f, "pointerdown", (function(e) {
      var t = q(e.target, "." + Aa) === u;
      if (bn(e, r, t)) {
        var i = !t && e.shiftKey;
        var v = function getHandleRect() {
          return qr(u);
        };
        var o = function getTrackRect() {
          return qr(f);
        };
        var l = function getHandleOffset(r, a) {
          return (r || v())[h] - (a || o())[h];
        };
        var g = w(n[c] || 0, 1 / wn(n)[p]);
        var b = e[s];
        var m = v();
        var S = o();
        var C = m[d];
        var x = l(m, S) + C / 2;
        var E = b - S[h];
        var P = t ? 0 : E - x;
        var T = [ Gr(a, "selectstart", (function(r) {
          return Zr(r);
        }), {
          C: false
        }), Gr(f, "pointermove", (function(r) {
          var a = r[s] - b;
          if (t || i) {
            g(P + a);
          }
        })) ];
        if (i) {
          g(P);
        } else if (!t) {
          var z = Fa()[de];
          if (z) {
            y(T, z.T(g, l, P, C, E));
          }
        }
        Gr(f, "pointerup", (function(r) {
          O(T);
          f.releasePointerCapture(r.pointerId);
        }), {
          P: true
        });
        f.setPointerCapture(e.pointerId);
      }
    }));
  };
  var Sn = function createScrollbarsSetupEvents(r, a) {
    return function(e, n, t, i, v, o) {
      var u = e.Wr;
      var f = wr(333), l = f[0], c = f[1];
      var s = !!v.scrollBy;
      var d = true;
      return O.bind(0, [ Gr(u, "pointerenter", (function() {
        n(Ra, true);
      })), Gr(u, "pointerleave pointercancel", (function() {
        n(Ra);
      })), Gr(u, "wheel", (function(r) {
        var a = r.deltaX, e = r.deltaY, t = r.deltaMode;
        if (s && d && 0 === t && F(u) === i) {
          v.scrollBy({
            left: a,
            top: e,
            behavior: "smooth"
          });
        }
        d = false;
        n(ka, true);
        l((function() {
          d = true;
          n(ka);
        }));
        Zr(r);
      }), {
        C: false,
        O: true
      }), mn(u, t), yn(r, t, e, v, a, o), c ]);
    };
  };
  var Cn = Math.min, On = Math.max, xn = Math.abs, En = Math.round;
  var Pn = function getScrollbarHandleLengthRatio(r, a, e, n) {
    if (n) {
      var t = e ? "x" : "y";
      var i = n.Pr, v = n.Er;
      var o = v[t];
      var u = i[t];
      return On(0, Cn(1, o / (o + u)));
    }
    var f = e ? "w" : "h";
    var l = Vr(r)[f];
    var c = Vr(a)[f];
    return On(0, Cn(1, l / c));
  };
  var Tn = function getScrollbarHandleOffsetRatio(r, a, e, n, t, i) {
    var v = ye(), o = v.B;
    var u = i ? "x" : "y";
    var f = i ? "Left" : "Top";
    var l = n.Pr;
    var c = En(l[u]);
    var s = xn(e["scroll" + f]);
    var d = i && t;
    var h = o.i ? s : c - s;
    var g = d ? h : s;
    var p = Cn(1, g / c);
    var w = Pn(r, a, i);
    return 1 / w * (1 - w) * p;
  };
  var zn = function createScrollbarsSetupElements(r, a, e) {
    var n = ye(), t = n.U;
    var i = t(), v = i.scrollbars;
    var o = v.slot;
    var u = a.ir, f = a.Z, l = a.$, s = a.J, d = a.lr, h = a.er;
    var g = d ? {} : r, p = g.scrollbars;
    var w = p || {}, b = w.slot;
    var m = Oe([ f, l, s ], (function() {
      return l;
    }), o, b);
    var S = function scrollbarStructureAddRemoveClass(r, a, e) {
      var n = e ? Pr : Er;
      each(r, (function(r) {
        n(r.Wr, a);
      }));
    };
    var x = function scrollbarsHandleStyle(r, a) {
      each(r, (function(r) {
        var e = a(r), n = e[0], t = e[1];
        style(n, t);
      }));
    };
    var E = function scrollbarStructureRefreshHandleLength(r, a, e) {
      x(r, (function(r) {
        var n;
        var t = r.Nr, i = r.Yr;
        return [ t, (n = {}, n[e ? "width" : "height"] = (100 * Pn(t, i, e, a)).toFixed(3) + "%", 
        n) ];
      }));
    };
    var P = function scrollbarStructureRefreshHandleOffset(r, a, e) {
      var n = e ? "X" : "Y";
      x(r, (function(r) {
        var t = r.Nr, i = r.Yr, v = r.Wr;
        var o = Tn(t, i, h, a, Mr(v), e);
        var u = o === o;
        return [ t, {
          transform: u ? "translate" + n + "(" + (100 * o).toFixed(3) + "%)" : ""
        } ];
      }));
    };
    var T = [];
    var z = [];
    var A = [];
    var L = function scrollbarsAddRemoveClass(r, a, e) {
      var n = c(e);
      var t = n ? e : true;
      var i = n ? !e : true;
      t && S(z, r, a);
      i && S(A, r, a);
    };
    var H = function refreshScrollbarsHandleLength(r) {
      E(z, r, true);
      E(A, r);
    };
    var M = function refreshScrollbarsHandleOffset(r) {
      P(z, r, true);
      P(A, r);
    };
    var R = function generateScrollbarDOM(r) {
      var a = r ? Pa : Ta;
      var n = r ? z : A;
      var t = C(n) ? Ma : "";
      var i = Z(xa + " " + a + " " + t);
      var v = Z(za);
      var o = Z(Aa);
      var f = {
        Wr: i,
        Yr: v,
        Nr: o
      };
      Y(i, v);
      Y(v, o);
      y(n, f);
      y(T, [ X.bind(0, i), e(f, L, u, l, h, r) ]);
      return f;
    };
    var D = R.bind(0, true);
    var I = R.bind(0, false);
    var k = function appendElements() {
      Y(m, z[0].Wr);
      Y(m, A[0].Wr);
      lr((function() {
        L(Ma);
      }), 300);
    };
    D();
    I();
    return [ {
      Gr: H,
      Xr: M,
      Zr: L,
      $r: {
        Jr: z,
        Kr: D,
        Qr: x.bind(0, z)
      },
      ra: {
        Jr: A,
        Kr: I,
        Qr: x.bind(0, A)
      }
    }, k, O.bind(0, T) ];
  };
  var An = function createScrollbarsSetup(r, a, e) {
    var n;
    var t;
    var i;
    var v;
    var o;
    var u = 0;
    var f = He({});
    var l = f[0];
    var c = wr(), s = c[0], d = c[1];
    var h = wr(), g = h[0], p = h[1];
    var w = wr(100), b = w[0], m = w[1];
    var y = wr(100), S = y[0], C = y[1];
    var x = wr((function() {
      return u;
    })), E = x[0], P = x[1];
    var T = zn(r, e.Ur, Sn(a, e)), z = T[0], A = T[1], L = T[2];
    var H = e.Ur, M = H.$, I = H.J, k = H.er, V = H.nr, j = H.cr, B = H.ur;
    var q = z.$r, U = z.ra, N = z.Zr, Y = z.Gr, W = z.Xr;
    var G = q.Qr;
    var X = U.Qr;
    var Z = function styleScrollbarPosition(r) {
      var a = r.Wr;
      var e = j && !B && F(a) === I && a;
      return [ e, {
        transform: e ? "translate(" + R(k) + "px, " + D(k) + "px)" : ""
      } ];
    };
    var $ = function manageScrollbarsAutoHide(r, a) {
      P();
      if (r) {
        N(Ia);
      } else {
        var e = function hide() {
          return N(Ia, true);
        };
        if (u > 0 && !a) {
          E(e);
        } else {
          e();
        }
      }
    };
    var J = function onHostMouseEnter() {
      v = t;
      v && $(true);
    };
    var K = [ m, P, C, p, d, L, Gr(M, "pointerover", J, {
      P: true
    }), Gr(M, "pointerenter", J), Gr(M, "pointerleave", (function() {
      v = false;
      t && $(false);
    })), Gr(M, "pointermove", (function() {
      n && s((function() {
        m();
        $(true);
        S((function() {
          n && $(false);
        }));
      }));
    })), Gr(V, "scroll", (function() {
      g((function() {
        W(e());
        i && $(true);
        b((function() {
          i && !v && $(false);
        }));
      }));
      j && G(Z);
      j && X(Z);
    })) ];
    var Q = l.bind(0);
    Q.Ur = z;
    Q.qr = A;
    return [ function(r, v, f) {
      var l = f.Ar, c = f.Lr, s = f.zr, d = f.br;
      var h = Le(a, r, v);
      var g = e();
      var p = g.Pr, w = g.Cr, b = g.wr;
      var m = h("scrollbars.theme"), y = m[0], S = m[1];
      var C = h("scrollbars.visibility"), O = C[0], x = C[1];
      var E = h("scrollbars.autoHide"), P = E[0], T = E[1];
      var z = h("scrollbars.autoHideDelay"), A = z[0];
      var L = h("scrollbars.dragScroll"), H = L[0], M = L[1];
      var R = h("scrollbars.clickScroll"), D = R[0], I = R[1];
      var k = l || c || d || v;
      var V = s || x || v;
      var j = function setScrollbarVisibility(r, a) {
        var e = "visible" === O || "auto" === O && "scroll" === r;
        N(La, e, a);
        return e;
      };
      u = A;
      if (S) {
        N(o);
        N(y, true);
        o = y;
      }
      if (T) {
        n = "move" === P;
        t = "leave" === P;
        i = "never" !== P;
        $(!i, true);
      }
      if (M) {
        N(ja, H);
      }
      if (I) {
        N(Va, D);
      }
      if (V) {
        var F = j(w.x, true);
        var q = j(w.y, false);
        var U = F && q;
        N(Ha, !U);
      }
      if (k) {
        Y(g);
        W(g);
        N(Da, !p.x, true);
        N(Da, !p.y, false);
        N(Ea, b && !B);
      }
    }, Q, O.bind(0, K) ];
  };
  var Ln = function invokePluginInstance(r, a, e) {
    if (s(r)) {
      r(a || void 0, e || void 0);
    }
  };
  var Hn = function OverlayScrollbars(r, a, e) {
    var n = ye(), t = n.Y, i = n.U, v = n.q;
    var o = Fa();
    var u = w(r);
    var f = u ? r : r.target;
    var l = ze(f);
    if (a && !l) {
      var c = false;
      var s = function validateOptions(r) {
        var a = Fa()[ae];
        var e = a && a.T;
        return e ? e(r, true) : r;
      };
      var d = P({}, t(), s(a));
      var h = Qr(e), g = h[0], p = h[1], b = h[2];
      var m = pn(r, d), y = m[0], S = m[1], C = m[2];
      var O = An(r, d, S), x = O[0], z = O[1], A = O[2];
      var L = function update(r, a) {
        return y(r, !!a);
      };
      var H = v(L.bind(0, {}, true));
      var M = function destroy(r) {
        Te(f);
        H();
        A();
        C();
        c = true;
        b("destroyed", [ R, !!r ]);
        p();
      };
      var R = {
        options: function options(r) {
          if (r) {
            var a = ea(d, s(r));
            if (!T(a)) {
              P(d, a);
              L(a);
            }
          }
          return P({}, d);
        },
        on: g,
        off: function off(r, a) {
          r && a && p(r, a);
        },
        state: function state() {
          var r = S(), a = r.Er, e = r.Pr, n = r.Cr, t = r.Tr, i = r.K, v = r.mr, o = r.wr;
          return P({}, {
            overflowEdge: a,
            overflowAmount: e,
            overflowStyle: n,
            hasOverflow: t,
            padding: i,
            paddingAbsolute: v,
            directionRTL: o,
            destroyed: c
          });
        },
        elements: function elements() {
          var r = S.Ur, a = r.Z, e = r.$, n = r.K, t = r.J, i = r.rr, v = r.er, o = r.nr;
          var u = z.Ur, f = u.$r, l = u.ra;
          var c = function translateScrollbarStructure(r) {
            var a = r.Nr, e = r.Yr, n = r.Wr;
            return {
              scrollbar: n,
              track: e,
              handle: a
            };
          };
          var s = function translateScrollbarsSetupElement(r) {
            var a = r.Jr, e = r.Kr;
            var n = c(a[0]);
            return P({}, n, {
              clone: function clone() {
                var r = c(e());
                x({}, true, {});
                return r;
              }
            });
          };
          return P({}, {
            target: a,
            host: e,
            padding: n || t,
            viewport: t,
            content: i || t,
            scrollOffsetElement: v,
            scrollEventElement: o,
            scrollbarHorizontal: s(f),
            scrollbarVertical: s(l)
          });
        },
        update: function update(r) {
          return L({}, r);
        },
        destroy: M.bind(0)
      };
      S.Fr((function(r, a, e) {
        x(a, e, r);
      }));
      Pe(f, R);
      each(E(o), (function(r) {
        return Ln(o[r], 0, R);
      }));
      if (xe(S.Ur.ur, i().cancel, !u && r.cancel)) {
        M(true);
        return R;
      }
      S.qr();
      z.qr();
      b("initialized", [ R ]);
      S.Fr((function(r, a, e) {
        var n = r.pr, t = r.br, i = r.gr, v = r.Ar, o = r.Lr, u = r.zr, f = r._r, l = r.Or;
        b("updated", [ R, {
          updateHints: {
            sizeChanged: n,
            directionChanged: t,
            heightIntrinsicChanged: i,
            overflowEdgeChanged: v,
            overflowAmountChanged: o,
            overflowStyleChanged: u,
            contentMutation: f,
            hostMutation: l
          },
          changedOptions: a,
          force: e
        } ]);
      }));
      R.update(true);
      return R;
    }
    return l;
  };
  Hn.plugin = function(r) {
    each(qa(r), (function(r) {
      return Ln(r, Hn);
    }));
  };
  Hn.valid = function(r) {
    var a = r && r.elements;
    var e = s(a) && a();
    return p(e) && !!ze(e.target);
  };
  Hn.env = function() {
    var r = ye(), a = r.j, e = r.H, n = r.L, t = r.B, i = r.F, v = r.M, o = r.G, u = r.X, f = r.U, l = r.N, c = r.Y, s = r.W;
    return P({}, {
      scrollbarsSize: a,
      scrollbarsOverlaid: e,
      scrollbarsHiding: n,
      rtlScrollBehavior: t,
      flexboxGlue: i,
      cssCustomProperties: v,
      staticDefaultInitialization: o,
      staticDefaultOptions: u,
      getDefaultInitialization: f,
      setDefaultInitialization: l,
      getDefaultOptions: c,
      setDefaultOptions: s
    });
  };
  r.ClickScrollPlugin = he;
  r.OverlayScrollbars = Hn;
  r.ScrollbarsHidingPlugin = se;
  r.SizeObserverPlugin = ie;
  Object.defineProperty(r, "v", {
    value: true
  });
  return r;
}({});
